docs/dev/apis/configurable-company.md

```diff
@@ -0,0 +1,142 @@
+
+## Requirements
+
+To develop with Configurable company, you should have the following mods:
+
+-   [BepinExPack by BepInEx](https://thunderstore.io/c/lethal-company/p/BepInEx/BepInExPack/)
+
+-   [ConfigurableCompany by Ansuz/AMRV](https://thunderstore.io/c/lethal-company/p/AMRV/ConfigurableCompany/)
+
+I encourage to use a [mod manager](https://thunderstore.io/c/lethal-company/p/ebkr/r2modman/) to install them and avoid any problems.
+
+## Setting up environment
+
+:::warning
+This guide assumes you are using [Visual studio](https://visualstudio.microsoft.com) as IDE.
+:::
+
+Before adding the dependency to your project you should have your base plugin created, if you don't know how see [Initial setup](/dev/initial-setup) and [Starting a mod](/dev/starting-a-mod).
+
+You can add the project via [NuGet packages](#using-nuget-package-manager) or by [referencing it manually](#adding-file-manually).
+
+### Using NuGet package manager
+
+If you wish to learn how to use **NuGet**, [here](https://learn.microsoft.com/en-us/nuget/quickstart/install-and-use-a-package-in-visual-studio) is a link to microsoft's guide.
+
+You need to add the package [**Amrv.ConfigurableCompany**](https://www.nuget.org/packages/Amrv.ConfigurableCompany). You can add it to your project by using the _manage nuget packages_ option from your project.  
+Otherwise you can add the reference by manually editing the `.csproj` file and adding the following tags:
+
+```xml
+<ItemGroup>
+    <PackageReference Include="Amrv.ConfigurableCompany" Version="2.5.0" />
+</ItemGroup>
+```
+
+### Adding dependency manually
+
+First of all you need a copy of the `.dll` file (wich you can [download manually](https://thunderstore.io/c/lethal-company/p/AMRV/ConfigurableCompany/versions/) or copy an existing one if you have the mod installed).
+
+Add the `Amrv.ConfigurableCompany.dll` to the library folder of your project. _Make sure you also include it as a reference in the project file `project.csproj`._
+
+:::info
+You can reference the full path to the dll that you are using for the game, however I don't recommend using absolute paths.
+:::
+
+```xml
+<ItemGroup>
+    <!--You can name the reference however you want as long as unique-->
+    <Reference Include="ConfigurableCompany">
+        <!--This is an example path, use the one you have with the dll-->
+        <HintPath>libs\ConfigurableCompany.dll</HintPath>
+    </Reference>
+</ItemGroup>
+```
+
+You might as well mark the mod as a hard dependency of your project.
+
+::: info
+You can get the dependency string and the mod version from the `LethalConfiguration` class.
+
+```cs
+    LethalConfiguration.PLUGIN_GUID
+    LethalConfiguration.PLUGIN_VERSION
+```
+
+:::
+
+```cs
+[BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
+[BepInDependency( // [!code ++]
+    LethalConfiguration.PLUGIN_GUID, // [!code ++]
+    BepInDependency.DependencyFlags.HardDependency)] // [!code ++]
```

Comment:
Spanning this across multiple lines probably isn't the best bet for readability.